pipenv/core.py

```diff
@@ -1503,14 +1501,6 @@ def pip_install(
                 extra_indexes = list(project.sources)
     if requirement and requirement.vcs or requirement.editable:
         requirement.index = None
-        # Install dependencies when a package is a non-editable VCS dependency.
-        # Don't specify a source directory when using --system.
-        if not requirement.editable and no_deps is not True:
```

Comment:
The comment indicated this first block was due to really really old Pipfile.lock and no tests fail around this.